Add comprehensive deployment pipeline documentation for agent-based s…

pavithrarao91 opened the pull request:
…ystems

- Introduced a new detailed guide on deployment architectures for agentic systems, covering four patterns from simple to complex.
- Included diagrams for each architecture pattern to enhance understanding.
- Expanded on configuration management, infrastructure management, and CI/CD pipeline strategies tailored for different organizational needs.
- Integrated best practices for deploying agents on Databricks, emphasizing governance, evaluation, and continuous improvement.
- Added new images to support the documentation and improve visual clarity.

pavithrarao91 committed:
Add comprehensive deployment pipeline documentation for agent-based systems

- Introduced a new detailed guide on deployment architectures for agentic systems, covering four patterns from simple to complex.
- Included diagrams for each architecture pattern to enhance understanding.
- Expanded on configuration management, infrastructure management, and CI/CD pipeline strategies tailored for different organizational needs.
- Integrated best practices for deploying agents on Databricks, emphasizing governance, evaluation, and continuous improvement.
- Added new images to support the documentation and improve visual clarity.

jeannefukumaru committed:
moved 1.3_deployment-pipeline-updated.qmd to part 1 folder to fix ebook rendering

jeannefukumaru committed:
Merge remote-tracking branch 'origin' into deployment_pipelines_new

jeannefukumaru commented on part1/1.2_deployment-pipeline-updated.qmd:
```diff
@@ -0,0 +1,318 @@
+#### Anatomy of a deployment pipeline
+
+[DRAFT lucidchart](https://lucid.app/lucidchart/3c5f3fc1-ad2a-40f6-9f89-74b6dbb1fe14/edit?invitationId=inv_07475903-1de7-457b-88d1-74adfe4f4ba2&page=t3zIPiA-na8H#)
```
It doesn't look like we are still using this. If not, can we remove it?

pavithrarao91 replied:
good catch - I will remove this part.

jeannefukumaru commented on part1/1.2_deployment-pipeline-updated.qmd:
```diff
@@ -0,0 +1,318 @@
+#### Anatomy of a deployment pipeline
```
I think the title needs to be updated to reflect the content of the chapter. Perhaps something like "Reference architecture patterns for agentic app deployments"?

jeannefukumaru commented on part1/1.2_deployment-pipeline-updated.qmd:
```diff
@@ -0,0 +1,318 @@
+
+The following sections analyze four deployment architecture patterns for agentic systems, progressing from simple to complex scenarios. Each pattern addresses different organizational needs around agent composition, account isolation, and multi-environment deployment strategies.
+
+##### Pattern 1: Multi-Environment Single-Account Single-Agent View
```
We should clarify for the reader what we define as "multi-envrionment", "single-account" and "multi-account" since we shouldn't assume that they are already familiar with these terms

jeannefukumaru commented:
@pavithrarao91 This is super detailed! Just some minor things:
- remove from diagrams that are not needed from the deployment-architecture_files and reference-architecture_files folders
- clarify some of the terminology used (outlined above in the comments) 
- the bullet formatting of the sections need to be fixed (you can check by rendering the book and chapters by running `quarto preview` in the IDE terminal
- remove the older chapter drafts from the part 1 folder

jeannefukumaru committed:
renamed files and fixed _quarto.yml so book renders correctly

pavithrarao91 committed:
Update deployment pipeline documentation and clean up resources
- Revised the section on deployment architecture patterns for agentic systems, enhancing clarity of the terminology used
- Removed outdated images deployment pipeline draft
- updated the formating and previwed for validations

pavithrarao91 commented:
@jeannefukumaru - please review again, I have made recommended changes.

jeannefukumaru committed:
Merge branch 'main' of github.com:dbczumar/ops into deployment_pipelines_new
